account_payment_order/models/account_move.py

```diff
@@ -39,6 +39,31 @@ def _compute_payment_order_ok(self):
                 payment_mode = move.payment_mode_id
             move.payment_order_ok = payment_mode.payment_order_ok
 
+    def _get_payment_order_communication(self):
+        """
+        Retrieve the communication string for the payment order
+        """
+        communication = self.ref or self.name or ""
```

Comment:
I think we should finally complement this this way:

```suggestion
        communication = self.payment_ref or self.ref or self.name or ""
```